sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py

```diff
@@ -116,6 +150,8 @@ class UnknownIdentifier(object):
     It will be encountered in communications with endpoints that are not
     identifiable by this version of the SDK.
 
+    It is not advisable to rely on this type of identifier, as UnknownIdentifier could become a new or existing distinct type in the future.
```

Comment:
More specifically, they should not rely on the `kind` property having the value `"unknown"`. 
The can, and should, still rely on the presence of the `raw_id` property, as this will not change even if the Identifies becomes a new "known" type layer.

Reply:
@annatisch before applying changes I want to get your feedback as I will be applying the same changes to all acs modalities. I rephrased it as follows:

> For checking yet unknown identifiers it is better to rely on the presence of the `raw_id` property, as new or existing distinct type identifiers always contain the `raw_id` property. It is not advisable to rely on the kind property with a value `unknown`, as it could become a new or existing distinct type in the future. 

If it sounds good and I will extend this [enum docstring](https://github.com/Azure/azure-sdk-for-python/blob/63bf4261c0c41c6bb44ea4b34f43f5ffb5cf88dc/sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py#L20) with the same comment

Reply:
but in the [identifier_from_raw_id](https://github.com/Azure/azure-sdk-for-python/blob/ca2dd99908bfd955357788e92824cb472d1c5885/sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py#L301) method the unknown identifier init with raw_id value, so does it really make sense to rely on it?

Reply:
I think the wording looks good - though I'm not sure I understand your question?
The `UnknownIdentifier` will still expose the raw_id value as an attribute - and that wont change once it becomes a new type later?

Reply:
Right now we provide a public API [identifier_from_raw_id](https://github.com/Azure/azure-sdk-for-python/blob/ca2dd99908bfd955357788e92824cb472d1c5885/sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py#L301) which creates an identifier type from raw id, and it can generate also UnknownIdentifier by initializing its raw_id property. So here Contoso could not rely on the raw_id property presence, as it is present in the UnknownIdentifier as well